src/command.js

```diff
@@ -185,13 +188,17 @@ export default class StorybookCommand extends Command {
       this.preview = { content: previewDOM, sha: sha256hash(previewDOM) };
       clearTimeout(logTimeout);
 
-      // borrow a browser page to discover stories
-      await this.percy.browser.launch();
       page = await this.percy.browser.page({ meta });
       await page.goto(previewUrl);
 
       /* istanbul ignore next: no instrumenting injected code */
-      return await page.eval((util, previewUrl) => {
+      return await page.eval(async ({ waitFor }, previewUrl) => {
+        // ensure the page has loaded and the var we need is present
+        await waitFor(() => !!window.__STORYBOOK_CLIENT_API__, 5000)
```

Comment:
Should the `5000` timeout be a constant at the top, shared with both evals?

We can pass the constant inside of eval by adding another arg, like `previewUrl` gets.... that arg does nothing. So you can rename that arg to `timeout` and pass in the constant as a second arg to `eval`.